core/common/src/main/java/alluxio/collections/IndexedSet.java

```diff
@@ -100,10 +100,8 @@
  */
 @ThreadSafe
 public class IndexedSet<T> extends AbstractSet<T> {
-  /** All objects in the set. This set is required to guarantee uniqueness of objects. */
-  // TODO(gpang): remove this set, and just use the indexes.
-  private final ConcurrentHashSet<T> mObjects = new ConcurrentHashSet<>(8, 0.95f, 8);
-
+  /** The first index. This is required to guarantee uniqueness of objects. */
```

Comment:
1 how about calling this primary index?
2 since it is required to guarantee uniqueness of objects, shall we check uniqueness in the constructor given the definition of this index?